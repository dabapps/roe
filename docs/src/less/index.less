@import '../../../src/less/index.less';

@body-background: @grey-lighter;

h1,
h2 {
  color: @secondary;
}

.container {
  box-shadow: @shadow-hard;
}

.display-columns {
  .column {
    background-color: @grey-lightest;
    border: @border-base;
  }
}
