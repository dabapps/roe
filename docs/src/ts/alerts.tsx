import * as React from 'react';

import {
  Alert,
  CodeBlock,
  Column,
  DabIpsum,
  Row,
  Section
} from '../../../src/ts';

const Alerts = () => (
  <Section>
    <Row>
      <Column>
        <h2>
          Alerts
        </h2>

        <Alert>
          <DabIpsum count={2} />
        </Alert>
        <Alert className="info">
          <DabIpsum count={2} />
        </Alert>
        <Alert className="success">
          <DabIpsum count={2} />
        </Alert>
        <Alert className="warning">
          <DabIpsum count={2} />
        </Alert>
        <Alert className="error">
          <DabIpsum count={2} />
        </Alert>
      </Column>
    </Row>
    <Row>
      <Column>
        <h3>
          Code
        </h3>
        <CodeBlock language="javascript" name="JSX">
          {`
            <Alert>
              <DabIpsum count={2} />
            </Alert>
            <Alert className="info">
              <DabIpsum count={2} />
            </Alert>
            <Alert className="success">
              <DabIpsum count={2} />
            </Alert>
            <Alert className="warning">
              <DabIpsum count={2} />
            </Alert>
            <Alert className="error">
              <DabIpsum count={2} />
            </Alert>
          `}
        </CodeBlock>
      </Column>
    </Row>
  </Section>
);

export default Alerts;
