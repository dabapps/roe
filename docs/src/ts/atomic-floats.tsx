import * as React from 'react';

import {
  CodeBlock,
  Column,
  Row,
  Section
} from '../../../src/ts';

const AtomicFloats = () => (
  <Section>
    <Row>
      <Column>
        <h2>
          Atomic Float Classes
        </h2>
      </Column>
    </Row>
    <Row>
      <Column className="display-columns">
        <h3>
          Demo
        </h3>
        <Row>
          <Column xs={6} className="xs-float-right sm-float-left md-float-right">
            float-right sm-float-left md-float-right
          </Column>
          <Column xs={6} className="xs-float-left sm-float-right md-float-left">
            float-left sm-float-right md-float-left
          </Column>
        </Row>
        <p>
          Note: float-left is an alias for xs-float-left, and float-right an alias for xs-float-right
        </p>
      </Column>
    </Row>
    <Row>
      <Column>
        <h3>
          Code
        </h3>
        <CodeBlock language="javascript">
          {`
            <Row>
              <Column xs={6} className="xs-float-right sm-float-left md-float-right">
                float-right sm-float-left md-float-right
              </Column>
              <Column xs={6} className="xs-float-left sm-float-right md-float-left">
                float-left sm-float-right md-float-left
              </Column>
            </Row>
          `}
        </CodeBlock>
      </Column>
    </Row>
  </Section>
);

export default AtomicFloats;
