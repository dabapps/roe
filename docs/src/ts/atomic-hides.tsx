import * as React from 'react';

import {
  CodeBlock,
  Column,
  Row,
  Section
} from '../../../src/ts';

const AtomicHides = () => (
  <Section>
    <Row>
      <Column>
        <h2>
          Atomic Hide & Display Classes
        </h2>
      </Column>
    </Row>
    <Row>
      <Column className="display-columns">
        <h3>
          Demo
        </h3>
        <Row>
          <Column xs={6} className="xs-display-none sm-display-block md-display-none">
            xs-display-none sm-display-block md-display-none
          </Column>
          <Column xs={6} className="xs-display-block sm-display-none md-display-block">
            xs-display-block sm-display-none md-display-block
          </Column>
        </Row>
        <p>
          Available display types are: none, block, inline, inline-block
        </p>
        <p>
          Note: display-none is an alias for xs-display-none, and display-block an alias for xs-display-block, etc
        </p>
      </Column>
    </Row>
    <Row>
      <Column>
        <h3>
          Code
        </h3>
        <CodeBlock language="javascript" name="JSX">
          {`
            <Row>
              <Column xs={6} className="xs-display-none sm-display-block md-display-none">
                xs-display-none sm-display-block md-display-none
              </Column>
              <Column xs={6} className="xs-display-block sm-display-none md-display-block">
                xs-display-block sm-display-none md-display-block
              </Column>
            </Row>
          `}
        </CodeBlock>
      </Column>
    </Row>
  </Section>
);

export default AtomicHides;
