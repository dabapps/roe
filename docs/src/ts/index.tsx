import * as React from 'react';
import * as ReactDOM from 'react-dom';

import './livereload';

import {
  Column,
  Container,
  Row
} from '../../../src/ts';

import { Buttons } from './buttons';
import { CodeBlocks } from './code-blocks';
import { Grid } from './grid';
import { Inputs } from './inputs';
import { Text } from './text';

const packageJson = require( '../../../package.json'); // tslint:disable-line:no-var-requires

class App extends React.Component<void, void> {
  public render () {
    return (
      <Container>
        <Row>
          <Column>
            <h1>
              Roe - DabApps' Project Development Kit (v{packageJson.version})
            </h1>
            <p>
              <strong>
                A Collection of React Components for Project Development
              </strong>
            </p>
          </Column>
        </Row>
        <Buttons />
        <CodeBlocks />
        <Grid />
        <Inputs />
        <Text />
      </Container>
    );
  }
}

ReactDOM.render(<App />, document.getElementById('app'));
