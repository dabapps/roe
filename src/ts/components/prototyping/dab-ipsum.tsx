import * as React from 'react';
import { PureComponent } from 'react';
import { generateIpsum } from '../../utils';
import { WORDS } from '../../words';

export interface DabIpsumProps {
  /**
   * Set the component to render a different element type.
   * @default "'p'"
   */
  component?: 'ul' | 'ol' | 'p' | 'text';
  /**
   * The number of elements (e.g. paragraphs) to render.
   * @default 5
   */
  count?: number;
}

const ipsumItem = (component: DabIpsumProps['component'], index: number) => {
  const ipsum = generateIpsum(WORDS);

  switch (component) {
    case 'ol':
    case 'ul':
      return (
        <li key={index}>
          {ipsum}
        </li>
      );
    case 'text':
      return (
        <span key={index}>
          {ipsum}
        </span>
      );
    // case 'p': NOTE: this is the default, so a case for it is not needed
    default:
      return (
        <p key={index}>
          {ipsum}
        </p>
      );
  }
};

/**
 * Custom Ipsum component, useful for rendering placeholder text when prototyping.
 */
export class DabIpsum extends PureComponent<DabIpsumProps, {}> {
  public shouldComponentUpdate (prevProps: DabIpsumProps) {
    return prevProps.component !== this.props.component ||
      prevProps.count !== this.props.count;
  }

  public render () {
    const {
      component = 'p',
      count = 5
    } = this.props;

    const items = Array.apply(null, new Array(count));

    switch (component) {
      case 'ul':
        return (
          <ul>
            {items.map((value: void, index: number) => ipsumItem(component, index))}
          </ul>
        );
      case 'ol':
        return (
          <ol>
            {items.map((value: void, index: number) => ipsumItem(component, index))}
          </ol>
        );
      case 'text':
        return (
          <span>
            {ipsumItem(component, 0)}
          </span>
        );
      // case 'p'
      default:
        return (
          <div>
            {items.map((value: void, index: number) => ipsumItem(component, index))}
          </div>
        );
    }
  }
}

export default DabIpsum;
