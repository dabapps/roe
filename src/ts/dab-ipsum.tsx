import * as randomSeed from 'random-seed';
import * as React from 'react';
import { HTMLProps, StatelessComponent } from 'react';

export interface IpsumProps {
  component: 'li' | 'p' | 'text';
}

export interface DabIpsumProps {
  component?: 'ul' | 'ol' | 'p' | 'text';
  count?: number;
}

const words = [
  'academia',
  'agile',
  'angular',
  'apprentice',
  'apps',
  'brighton',
  'business',
  'careers',
  'cats',
  'client',
  'coffee',
  'community',
  'css',
  'dabapps',
  'design',
  'designers',
  'development',
  'developers',
  'digital',
  'django',
  'documentation',
  'education',
  'equality',
  'events',
  'framework',
  'green',
  'innovation',
  'ionic',
  'ios',
  'java',
  'javascript',
  'kittens',
  'llamas',
  'mobile',
  'native',
  'networking',
  'news',
  'objective-c',
  'open-source',
  'partnerships',
  'product',
  'projects',
  'prototyping',
  'python',
  'react',
  'react-native',
  'recognition',
  'rest',
  'service',
  'smashing',
  'tea',
  'teaching',
  'technical',
  'testimonial',
  'testing',
  'toolkit',
  'training',
  'typescript',
  'web'
];

let rand = randomSeed.create('dabapps');

export const resetRandomSeed = () => {
  rand = randomSeed.create('dabapps');
}

export const generateIpsum = () => {
  const ipsum = Array.apply(null, new Array(15)).map(() => (
    words[Math.floor(rand.range(words.length))]
  )).join(' ');

  return ipsum.charAt(0).toUpperCase() + ipsum.substring(1) + '.';
}

const Ipsum: StatelessComponent<IpsumProps> = (props) => {
  const { component } = props;
  const ipsum = generateIpsum();

  switch (component) {
    case 'li':
      return (
        <li>
          {ipsum}
        </li>
      );
    case 'text':
      return (
        <span>
          {ipsum}
        </span>
      );
    // case 'p': NOTE: this is the default, so a case for it is not needed
    default:
      return (
        <p>
          {ipsum}
        </p>
      );
  }
}

// tslint:disable-next-line:max-classes-per-file
export class DabIpsum extends React.Component<DabIpsumProps, void> {
  public shouldComponentUpdate (prevProps: DabIpsumProps) {
    return prevProps.component !== this.props.component ||
      prevProps.count !== this.props.count;
  }

  public render () {
    const {
      component = 'p',
      count = 5
    } = this.props;

    const items = Array.apply(null, new Array(count)).map((v: void, index: number) => (
      <Ipsum key={index} component={component === 'p' ? component : 'li'} />
    ));

    switch (component) {
      case 'ul':
        return (
          <ul>
            {items}
          </ul>
        );
      case 'ol':
        return (
          <ol>
            {items}
          </ol>
        );
      case 'text':
        return (
          <Ipsum component="text" />
        );
      // case 'p': NOTE: this is the default, so a case for it is not needed
      default:
        return (
          <div>
            {items}
          </div>
        );
    }
  }
}
