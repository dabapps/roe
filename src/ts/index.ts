export { default as Alert } from './components/misc/alert';
export { default as AppRoot } from './components/app/root';
export { default as Anchor } from './components/misc/anchor';
export { default as Badge } from './components/misc/badge';
export { default as Banner } from './components/banners/banner';
export { default as Button } from './components/forms/button';
export { default as CodeBlock } from './components/misc/code-block';
export { default as Collapse } from './components/misc/collapse';
export { default as Column } from './components/grid/column';
export { default as Container } from './components/grid/container';
export { default as ContentBox } from './components/content/content-box';
export { default as ContentBoxHeader } from './components/content/content-box-header';
export { default as ContentBoxFooter } from './components/content/content-box-footer';
export { default as CookieBanner } from './components/banners/cookie-banner';
export { CookieBannerRenderProps } from './components/banners/cookie-banner';
export { default as DabIpsum } from './components/prototyping/dab-ipsum';
export { default as Footer } from './components/navigation/footer';
export { default as FormGroup } from './components/forms/form-group';
export { default as Highlight } from './components/misc/highlight';
export { default as InputGroup } from './components/forms/input-group';
export { default as InputGroupAddon } from './components/forms/input-group-addon';
export { default as InputWithPrefixSuffix } from './components/precomposed/input-with-prefix-suffix';
export { default as Modal } from './components/modals/modal';
export { default as ModalBody } from './components/modals/modal-body';
export { default as ModalCloseIcon } from './components/modals/modal-close-icon';
export { default as ModalFooter } from './components/modals/modal-footer';
export { default as ModalHeader } from './components/modals/modal-header';
export { default as ModalRenderer } from './components/modals/modal-renderer';
export { default as Nav } from './components/navigation/nav';
export { default as NavBar } from './components/navigation/nav-bar';
export { default as NavItem } from './components/navigation/nav-item';
export { default as Row } from './components/grid/row';
export { default as Section } from './components/content/section';
export { default as SideBar } from './components/navigation/side-bar';
export { default as SpacedGroup } from './components/misc/spaced-group';
export { default as SpeechBubble } from './components/misc/speech-bubble';
export { default as Tabs } from './components/tabs/tabs';
export { default as Tab } from './components/tabs/tab';
export { default as Table } from './components/tables/table';
export { default as TableBody } from './components/tables/table-body';
export { default as TableCell } from './components/tables/table-cell';
export { default as TableHead } from './components/tables/table-head';
export { default as TableHeader } from './components/tables/table-header';
export { default as TableRow } from './components/tables/table-row';
export { default as Well } from './components/misc/well';
