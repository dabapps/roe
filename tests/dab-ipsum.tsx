import { expect } from 'chai';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { DabIpsum } from '../src/ts/';
import { generateIpsum, resetRandomSeed } from '../src/ts/dab-ipsum';

describe('DabIpsum', () => {

  beforeEach(() => {
    resetRandomSeed();
  })

  it('should render some paragraphs by default', () => {
    const tree = renderer.create(
      <DabIpsum />
    );

    expect(tree).to.matchSnapshot();
  });

  it('should allow setting how many items to render', () => {
    const tree = renderer.create(
      <DabIpsum count={3} />
    );

    expect(tree).to.matchSnapshot();
  });

  it('should allow rendering an unordered list', () => {
    const tree = renderer.create(
      <DabIpsum type="ul" />
    );

    expect(tree).to.matchSnapshot();
  });

  it('should allow rendering an ordered list', () => {
    const tree = renderer.create(
      <DabIpsum type="ol" />
    );

    expect(tree).to.matchSnapshot();
  });

  it('should allow rendering plain text (in a span)', () => {
    const tree = renderer.create(
      <DabIpsum type="text" />
    );

    expect(tree).to.matchSnapshot();
  });

  it('should generate some ipsum', () => {
    const ipsum = generateIpsum();

    expect(ipsum).to.matchSnapshot();
  })

});
