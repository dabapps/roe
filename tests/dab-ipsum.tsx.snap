exports[`DabIpsum should allow rendering an ordered list 1`] = `
<ol>
  <li>
    Teaching javascript client agile developers apps training partnerships product training equality teaching business careers javascript.
  </li>
  <li>
    Objective-c testimonial academia product designers client training apprentice brighton partnerships testimonial django react teaching designers.
  </li>
  <li>
    Prototyping teaching angular client designers rest news testimonial academia prototyping designers testimonial apprentice equality web.
  </li>
  <li>
    Development framework technical ionic rest networking testing technical web testimonial cats agile digital rest django.
  </li>
  <li>
    Design toolkit networking mobile rest react-native django ionic product kittens testing react product networking prototyping.
  </li>
</ol>
`;

exports[`DabIpsum should allow rendering an unordered list 1`] = `
<ul>
  <li>
    Teaching javascript client agile developers apps training partnerships product training equality teaching business careers javascript.
  </li>
  <li>
    Objective-c testimonial academia product designers client training apprentice brighton partnerships testimonial django react teaching designers.
  </li>
  <li>
    Prototyping teaching angular client designers rest news testimonial academia prototyping designers testimonial apprentice equality web.
  </li>
  <li>
    Development framework technical ionic rest networking testing technical web testimonial cats agile digital rest django.
  </li>
  <li>
    Design toolkit networking mobile rest react-native django ionic product kittens testing react product networking prototyping.
  </li>
</ul>
`;

exports[`DabIpsum should allow rendering plain text (in a span) 1`] = `
<span>
  Teaching javascript client agile developers apps training partnerships product training equality teaching business careers javascript.
</span>
`;

exports[`DabIpsum should allow setting how many items to render 1`] = `
<div>
  <p>
    Teaching javascript client agile developers apps training partnerships product training equality teaching business careers javascript.
  </p>
  <p>
    Objective-c testimonial academia product designers client training apprentice brighton partnerships testimonial django react teaching designers.
  </p>
  <p>
    Prototyping teaching angular client designers rest news testimonial academia prototyping designers testimonial apprentice equality web.
  </p>
</div>
`;

exports[`DabIpsum should generate some ipsum 1`] = `"Teaching javascript client agile developers apps training partnerships product training equality teaching business careers javascript."`;

exports[`DabIpsum should render some paragraphs by default 1`] = `
<div>
  <p>
    Teaching javascript client agile developers apps training partnerships product training equality teaching business careers javascript.
  </p>
  <p>
    Objective-c testimonial academia product designers client training apprentice brighton partnerships testimonial django react teaching designers.
  </p>
  <p>
    Prototyping teaching angular client designers rest news testimonial academia prototyping designers testimonial apprentice equality web.
  </p>
  <p>
    Development framework technical ionic rest networking testing technical web testimonial cats agile digital rest django.
  </p>
  <p>
    Design toolkit networking mobile rest react-native django ionic product kittens testing react product networking prototyping.
  </p>
</div>
`;
